svetlana_kurakina_ndvi.py

```diff
@@ -0,0 +1,904 @@
+# harv_band_5 = rxr.open_rasterio(path_harv_bands[1], masked=True).squeeze()
+
+# # Clip band 5
+# harv_band_5_crop = harv_band_5.rio.clip(harv_crop_bound.geometry)
+
+# # Specify the valid range for band 5
+
+# if valid_range:
+#     mask = ((
+#         harv_band_5_crop < valid_range[0]) | (
+#         harv_band_5_crop > valid_range[1]))
+#     harv_band_5_crop = harv_band_5_crop.where(~xr.where(mask, True, False))
+    
+# #harv_band_5_crop.plot()
+
+# # Calculate NDVI (since I used cropped bands it is already cropped)
+# harv_ndvi = (harv_band_5_crop-harv_band_4_crop)/(
+#     harv_band_5_crop+harv_band_4_crop)
+
+# # Grab cloud values using earthpy
+# high_cloud_confidence = (
+#     em.pixel_flags["pixel_qa"]["L8"]["High Cloud Confidence"])
+# cloud = em.pixel_flags["pixel_qa"]["L8"]["Cloud"]
+# cloud_shadow = em.pixel_flags["pixel_qa"]["L8"]["Cloud Shadow"]
+# all_masked_values = cloud_shadow + cloud + high_cloud_confidence
+
+# # Open and crop harv qa layer
+# harv_qa = rxr.open_rasterio(path_harv_qa).squeeze()
+# harv_qa_cropped = harv_qa.rio.clip(harv_crop_bound.geometry)
+
+# Create and apply cloud mask to NDVI
+# harv_ndvi_cl_free = harv_ndvi.where(
+#     ~harv_qa_cropped.isin(all_masked_values))
+#harv_ndvi_cl_free.plot()
+
+# # Calculate mean NDVI
+# print('My mean NDVI for HARV site is:', harv_ndvi_cl_free.mean().values)
+
+# #Get site name from path
+# harv_sitename = [os.path.basename(os.path.normpath(path_harv))]
+
+# # Get date 
+# harv_date = [os.path.basename(os.path.normpath(path_harv_scene))[10:18]]
+
+# # Get NDVI value
+# harv_mean = [harv_ndvi_cl_free.mean().values]
+
+# # Construct dataframe
+# harv_df = pd.DataFrame(columns=["site","date","mean_ndvi"])
+# harv_df['site'] = harv_sitename
+# harv_df['date'] = harv_date
+# harv_df['mean_ndvi'] = harv_mean
+# harv_df['date'] = pd.to_datetime(harv_df['date'])
+# harv_df.set_index('date', inplace=True)
+# harv_df
+
+
+# ## Task 1: 
+# 
+# In the cell below, create a single dataframe containing MEAN NDVI, the site name, 
+# and the date of the data for the HARV site 
+# scene `HARV/landsat-crop/LC080130302017031701T1-SC20181023151837`.  The column names for the  final
+# DataFrame should be`mean_ndvi`, and `site`, and the data should be **indexed on the date**. 
+# 
+# Use the functions that we reviewed in class (or create your own versions of them) to implement your code
+# 
+# ### In the Cell below Place  All Functions Needed to Run this Notebook (20 points)
+
+# In[5]:
+
+
+### DO NOT REMOVE THIS LINE OR EDIT / MOVE THIS CELL ###
+start_time = datetime.now()
+
+
+# In[6]:
+
+
+# In this cell place all of the functions needed to run your notebook
+# You will be graded here on function application, docstrings, efficiency so ensure
+# All functions are placed here!
+
+# YOUR CODE HERE
+
+# Define function which opens, crops, and cleans landsat bands
+def open_clean_bands(band_path,
+                     crop_extent,
+                     valid_range=None,):
```

Comment:
Should '=' have spaces around it? There are a few more instances I've spotted in the code.

Reply:
So far I was sticking with the following approach: if I am making a new variable, I am using spaces around '=', when I specify input values for a function I do not use spaces. However, you made me thinking and I decided to go check the documentation and discussions on that. Here is waht i was able to find [Decumentation](https://www.python.org/dev/peps/pep-0008/#other-recommendations),  [Stackoverflow discussion](https://stackoverflow.com/questions/8853063/pep-8-why-no-spaces-around-in-keyword-argument-or-a-default-parameter-value). So i guess having no spaces around a default value in a function is kinda correct. I think i will ask in class or maybe post the question on canvas to make sure :)